Library/Homebrew/cmd/upgrade.rb

```diff
@@ -171,4 +181,100 @@ def upgrade_formula(f)
       nil
     end
   end
+
+  def relink_dependents(formulae)
+    return if formulae.empty?
+
+    kegs = Formula.installed
+                  .map(&:opt_or_installed_prefix_keg)
+                  .reject { |f| f.runtime_dependencies.to_a.empty? }
+
+    formulae_to_assess = formulae
+    wave = 0
+
+    until formulae_to_assess.empty?
+      wave += 1
+
+      oh1 "Checking formulae for broken library links - wave #{wave}" if ARGV.verbose?
+
+      formulae_to_reinstall = []
+      formulae_to_upgrade = []
+
+      CacheStoreDatabase.use(:linkage) do |db|
+        formulae_to_assess.each do |formula|
+          dependents = kegs.keep_if do |f|
+            f.runtime_dependencies
+             .any? { |d| d["full_name"] == formula.full_name }
+          end
+
+          next if dependents.empty?
+
+          ohai "Checking dependents of #{formula}" if ARGV.verbose?
+
+          broken_dependents = []
+
+          dependents.each do |keg|
+            checker = LinkageChecker.new(keg, cache_db: db)
+            broken_dependents << Formulary.from_keg(keg) if checker.broken_library_linkage?
+          end
+
+          next if broken_dependents.empty?
+
+          ohai "Dependents of #{formula} have broken dynamic library links!"
+
+          upgrade_possible, from_source = broken_dependents.partition do |f|
+            f.outdated?(fetch_head: ARGV.fetch_head?)
+          end
+
+          upgradable, pinned = upgrade_possible.partition(&:pinned?)
+
+          if !pinned.empty? && !ARGV.include?("--ignore-pinned")
+            oh1 "Skipping #{Formatter.pluralize(pinned.length, "pinned dependent")}:"
+            puts pinned.map { |f| "#{f.full_specified_name} #{f.pkg_version}" } * ", "
+          end
+
+          unless upgradable.empty?
+            oh1 "Upgrading #{Formatter.pluralize(upgradable.length, "dependent")}:"
```

Comment:
I reckon this should be printed just before the upgrade happens

Reply:
Please see the explanation above.